packages/dataviews/src/test/dataform.tsx

```diff
@@ -0,0 +1,346 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import Dataform from '../components/dataform/index';
+
+const fields = [
+	{
+		id: 'title',
+		label: 'Title',
+		type: 'text' as const,
+	},
+	{
+		id: 'order',
+		label: 'Order',
+		type: 'integer' as const,
+	},
+	{
+		id: 'author',
+		label: 'Author',
+		type: 'integer' as const,
+		elements: [
+			{ value: 1, label: 'Jane' },
+			{ value: 2, label: 'John' },
+		],
+	},
+];
+
+const form = {
+	fields: [ 'title', 'order', 'author' ],
+};
+
+const data = {
+	title: 'Hello World',
+	author: 1,
+	order: 1,
+};
+
+const fieldsSelector = {
+	title: {
+		view: () =>
+			screen.getByRole( 'button', {
+				name: /edit title/i,
+			} ),
+		edit: () =>
+			screen.getByRole( 'textbox', {
+				name: /title/i,
+			} ),
+	},
+	author: {
+		view: () =>
+			screen.getByRole( 'button', {
+				name: /edit author/i,
+			} ),
+		edit: () =>
+			screen.queryByRole( 'combobox', {
+				name: /author/i,
+			} ),
+	},
+	order: {
+		view: () =>
+			screen.getByRole( 'button', {
+				name: /edit order/i,
+			} ),
+		edit: () =>
+			screen.getByRole( 'spinbutton', {
+				name: /order/i,
+			} ),
+	},
+};
+
+describe( 'DataForm component', () => {
+	describe( 'in regular mode', () => {
+		it( 'should display fields', () => {
+			render(
+				<Dataform
+					onChange={ () => void 0 }
```

Comment:
Why not just using `() => {}` as a noop? The `void` operator is not very common in our codebase.

Reply:
It makes sense. Fixed with [238f654](https://github.com/WordPress/gutenberg/pull/68054/commits/238f654931b5e50ed1bc1f21ca1557964d4927a7).